File: .github/roundtrip/roundtrip.js
const fs = require('fs'), path = require('path');
const n = require('/tmp/nsv-js/index.js');
const dir = process.argv[2];
const files = fs.readdirSync(dir).filter(f => f.endsWith('.nsv')).sort();
let passed = 0; const fails = [];
for (const f of files) {
    const p = path.join(dir, f), orig = fs.readFileSync(p, 'utf8');
    n.stringify(n.parse(orig)) === orig ? passed++ : fails.push(f);
}
console.log(`  ${passed}/${passed + fails.length} passed`);
fails.forEach(f => console.log(`  ${f}`));
if (fails.length) process.exit(1);

File: .github/roundtrip/roundtrip.py
import os, sys, nsv

d = sys.argv[1]
entries = sorted(e for e in os.listdir(d) if e.endswith('.nsv'))
fails = [n for n in entries
         if nsv.dumps(nsv.loads(open(os.path.join(d, n)).read()))
            != open(os.path.join(d, n)).read()]
passed = len(entries) - len(fails)
print(f'  {passed}/{len(entries)} passed')
for f in fails: print(f'  {f}')
sys.exit(1 if fails else 0)
